server/src/main/java/org/elasticsearch/Version.java

```diff
@@ -504,7 +508,25 @@ private static class DeclaredVersionsHolder {
      * is a beta or RC release then the version itself is returned.
      */
     public Version minimumCompatibilityVersion() {
-        return V_5_0_0;
+        Version res = minCompatVersion;
+        if (res == null) {
+            res = computeMinCompatVersion();
+            minCompatVersion = res;
+        }
+        return res;
+    }
+
+    private Version computeMinCompatVersion() {
+        if (major < 7) {
+            // before CrateDB 4.0.0, version and minimum compatibility version were different, let's return the version
+            return this;
```

Comment:
I'm a bit confused about the comment, if version and minimum compatibility version were different, isn't returning `this` wrong as it implies version==minCompatVersion ?

Reply:
Right, with "different" I meant: "different semantic" as I thought we used something different but actually it was also pointing to some ES version compatible with our nowadays ones. Will look up what was used here and return it.